src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1837,15 +1837,23 @@ protected function prepareRequestUri()
         } elseif ($this->server->has('REQUEST_URI')) {
             $requestUri = $this->server->get('REQUEST_URI');
 
-            // HTTP proxy reqs setup request URI with scheme and host [and port] + the URL path, only use URL path
-            $uriComponents = parse_url($requestUri);
+            if ('' !== $requestUri && '/' === $requestUri[0]) {
+                // To only use path and query remove the fragment.
+                if (false !== $pos = strpos($requestUri, '#')) {
```

Comment:
Are there any cases were the fragment is embedded in the request URI?

Reply:
From all functional test I was able to do (e.g. nginx and apache2) I did not got the fragment and any absolute URL. That's expected as the fragment is not shared by the browser to the server.

Then it left cases when the request URI is set internally, if it is. And maybe very specific case with nginx, apache and/or other system.

@thomasbisignani @nkl1996 @nicolas-grekas Do you have more information when the URL is not absolute?

---

The main goal of this is to avoid regression regarding previous usage of `parse_url()`.